codebuild/codebuild_ECR_push/code_build_ecr_container.yml

```diff
@@ -3,17 +3,17 @@ version: 0.2
 phases:
   pre_build:
     commands:
+      - echo Assuming role into target AWS account for deployment
+      - role_arn="arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ROLE_NAME}"
+      - CODEPIPELINE_DEPLOYER_ROLE=`aws sts assume-role --role-arn $role_arn --role-session-name codepipeline-ecr-task`
+      - export AWS_ACCESS_KEY_ID=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.AccessKeyId'`
+      - export AWS_SECRET_ACCESS_KEY=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.SecretAccessKey'`
+      - export AWS_SESSION_TOKEN=`echo "${CODEPIPELINE_DEPLOYER_ROLE}" | jq -r '.Credentials.SessionToken'`
```

Comment:
Could all of this assume role code be moved into a script that can be stored on the base image and shared between modules?

Reply:
Good shout, can do 👍

Reply:
It's already there. It's in `bin/assume-role.sh`